ibis/expr/operations/relations.py

```diff
@@ -785,3 +786,36 @@ class NotExistsSubquery(Node):
 
     def output_type(self):
         return ir.ExistsExpr
+
+
+@public
+class FillNa(TableNode, sch.HasSchema):
+    """Fill null values in the table."""
+
+    table = Arg(rlz.table)
+    replacements = Arg(
+        rlz.one_of(
+            (
+                rlz.numeric,
+                rlz.string,
```

Comment:
Couple of questions :) I think the non-mapping behavior is pretty fraught.

* What is the behavior of `rlz.numeric`/`rlz.string` where one or more columns do not match the type of the value provided?
* Similarly, what if the replacement value cannot be safely promoted to one or more column types?
* Does the operation fail in those cases? If so, does it fail at ibis-compile-time or is it backend specific?
* Is the behavior when `replacements` is a column expression equivalent to `coalesce`?

Reply:
The behavior when the user passes a value whose type does not match one or more of the columns is backend-specific - for example, `df.fillna('string_value')` - Pandas will successfully fill float columns with 'string_value', while Pyspark will ignore. Or, `df.fillna(False)` - Pandas will fill float columns with False, while Pyspark will fill with 0.0. I'm not aware of a case where `type(value) != df[column].dtype` will actually fail; the individual backend will either box the replacement value into the column type, promote it, or ignore it.

At the same time, `t.fillna(val)` is just syntactic sugar for `t.fillna({c: val for c in t.columns})`, so unless we commit to doing some internal ibis validation at compile time, I am not sure why the syntactic sugar version of the API is problematic compared to `t.fillna({c: val for c in t.columns})`.

Also for the final point, I don't think we need to support replacements as a column expression if we already have coalesce.

Reply:
> At the same time, t.fillna(val) is just syntactic sugar for t.fillna({c: val for c in t.columns}), so unless we commit to doing some internal ibis validation at compile time, I am not sure why the syntactic sugar version of the API is problematic compared to t.fillna({c: val for c in t.columns}).

Good point. I think I am just bikeshedding on whether we should do some kind of type checking here, but this isn't blocking the PR.

Reply:
Yeah it's definitely something to consider, if the goal is to achieve as close to 100% consistent behavior across all backends then we'd be constrained by the most restrictive backend behavior which seems to be Pyspark at this point.